docs/manual/adapting-existing.md

```diff
@@ -71,17 +71,39 @@ directory.
 
 ## Booting and initramfs technology
 
-OSTree comes with optional dracut+systemd integration code that parses
-the `ostree=` kernel command line argument in the initramfs, and then
-sets up the read-only bind mount on `/usr`, a bind mount on the
-deployment's `/sysroot` to the physical `/`, and then finally uses
-`mount(MS_MOVE)` to make the deployment root appear to be the root
-filesystem before telling systemd to switch root.
+OSTree comes with optional dracut+systemd integration code:
+- parse the `ostree=` kernel command line argument in the initramfs
+- set up the read-only bind mount on `/usr` 
+- bind mount the deployment's `/sysroot` to the physical `/`
+- use `mount(MS_MOVE)` to make the deployment root appear to be the 
+root filesystem
+After these steps, systemd switches root.
 
 If you are not using dracut or systemd, using OSTree should still be
-possible, but you will have to write the integration code.  Patches to
-support other initramfs technologies and init systems, if sufficiently
-clean, will likely be accepted upstream.
+possible, but you will have to write the integration code. E.g:
+```bash
```

Comment:
How about:

> but you will have to write the integration code.  See the existing
> sources in `src/switchroot` as a reference, as well as `src/switchroot/switchroot.sh`.

(And copy the code you have to `src/switchroot/switchroot.sh`)

?